ha/link.c

```diff
@@ -1717,7 +1717,7 @@ M0_INTERNAL void m0_ha_link_rpc_endpoint(struct m0_ha_link *hl,
                                          m0_bcount_t        buf_len)
 {
 	m0_mutex_lock(&hl->hln_lock);
-	strncpy(buf, hl->hln_conn_cfg.hlcc_rpc_endpoint, buf_len);
+	strncpy(buf, hl->hln_conn_cfg.hlcc_rpc_endpoint, buf_len - 1);
```

Comment:
Does the compiler have a stricter checking on this 'strncpy()'?
Or is this a potential bug in the original code?

Reply:
Looks like it does, yes. See the errors in the commit msgs.